Mindustry is missing `zenity` dependency

### Describe the bug
A clear and concise description of what the bug is.

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-shell -p mindustry`
2. in main menu go to `settings` then `game data`
3. click `Export save`

### Additional context
```
[nix-shell:~/nixpkgs]$ mindustry
[I] Initialized Discord rich presence.
[I] [Core] Initialized SDL v2.30.3
[I] [Audio] Initialized SoLoud 202111 using MiniAudio at 44100hz / 900 samples / 2 channels
[I] [GL] Version: OpenGL 4.6.0 / AMD / AMD Radeon Graphics (radeonsi, renoir, LLVM 17.0.6, DRM 3.54, 6.6.36)
[I] [GL] Max texture size: 16384
[I] [GL] Using OpenGL 2 context.
[I] [JAVA] Version: 17.0.11
[I] [RAM] Available: 3.7 GB
[I] [Mindustry] Version: 146
[I] Total time to load: 2176ms
[I] Fetching community servers at https://raw.githubusercontent.com/Anuken/Mindustry/master/servers_v7.json
[I] Fetched 83 community servers.
[E] java.lang.RuntimeException: java.io.IOException: Cannot run program "zenity": error=2, No such file or directory
	at arc.util.OS.exec(OS.java:101)
	at arc.util.OS.exec(OS.java:107)
	at mindustry.core.Platform.lambda$showZenity$6(Platform.java:174)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.io.IOException: Cannot run program "zenity": error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
	at java.base/java.lang.Runtime.exec(Runtime.java:594)
	at java.base/java.lang.Runtime.exec(Runtime.java:453)
	at arc.util.OS.exec(OS.java:82)
	... 3 more
Caused by: java.io.IOException: error=2, No such file or directory
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
	... 7 more

[W] zenity not found, using non-native file dialog. Consider installing `zenity` for native file dialogs.
```

### Notify maintainers

@chkno @fgaz @thekostins
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.11 (Vicuna), 24.11.20240703.9f4128e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos-23.05"`
 - channels(sigmanificient): `""`
 - nixpkgs: `/nix/store/dk2rpyb6ndvfbf19bkb2plcz5y3k8i5v-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/games/mindustry/default.nix
+++ b/pkgs/games/mindustry/default.nix
@@ -7,6 +7,7 @@
 , fetchFromGitHub
 , gradle
 , jdk17
+, zenity
 
 # for arc
 , SDL2
@@ -189,6 +190,7 @@ stdenv.mkDerivation {
       mkdir -p $out/bin
       makeWrapper ${jdk}/bin/java $out/bin/mindustry \
         --add-flags "-jar $out/share/mindustry.jar" \
+        ${lib.optionalString stdenv.isLinux "--suffix PATH : ${lib.makeBinPath [zenity]}"} \
         --suffix LD_LIBRARY_PATH : ${lib.makeLibraryPath [libpulseaudio alsa-lib libjack2]} \
         --set ALSA_PLUGIN_DIR ${alsa-plugins}/lib/alsa-lib/'' + optionalString enableWayland '' \
         --set SDL_VIDEODRIVER wayland \

